---
title: "20201201_OS_radierant_orthologs"
author: "Matt Paul"
date: "`r format(Sys.Date(), '%m/%d/%Y')`"
output:  
  html_document:
    code_folding: hide
    number_sections: TRUE
    theme: yeti
    highlight: tango
    toc: true
    toc_float: true
---

# Plan

Repeat comparison but include bombyx. 

compare and contrast the moori datasets. look for overlaps etc. 

Compare to d melanogaster.
then also comapre across in both directiosns.

Look for overlaps. 


retry refafctored in simplest way so i can post on github

```{r}
require(magrittr)
require(tidyverse)
require(ShortRead)
require(Herper)

library(BiocParallel)
library(batchtools)

template <- "/rugpfs/fs0/brc/scratch/mpaul/bt-simple.tmpl"
param <- BatchtoolsParam(workers=10000, cluster="slurm", template=template)
register(param)


```

```{r getSeqFromUniprot, eval =F}
require(UniProt.ws)
taxon <- availableUniprotSpecies(pattern="Bombyx")
bombyx <- UniProt.ws(7091)
allUniprotIDs <- keys(bombyx ,keytype = "UNIPROTKB")
allInfoColumns <- keytypes(bombyx )
grep("UNIPROTKB",
     allInfoColumns)

#bgTable <- select(bombyx ,keys=allUniprotIDs,columns = allInfoColumns,keytype = "UNIPROTKB")
bgTable <- select(bombyx ,keys=allUniprotIDs,columns = c("SEQUENCE"),keytype = "UNIPROTKB")
require(Biostrings)
seqToWrite <- AAStringSet(bgTable$SEQUENCE)
names(seqToWrite) <- bgTable$UNIPROTKB
writeXStringSet(seqToWrite,filepath = "/rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/bombyx.fa")

```


```{R, eval=F}

file.copy("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa", "obiroi.fa")
o_fa<-readFasta("obiroi.fa")

require(Herper)
export_CondaEnv("eggnog", yml_export= "eggnog_tom.yml", file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))
import_CondaEnv("eggnog_tom.yml", "eggnog", pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda")

#install_CondaTools("ete3_external_apps","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper/eggnog",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)
#external apps is not installed. this gives warning in build check. unclear why or if needed. 

Herper::with_CondaEnv("eggnog",system("ete3 build check"),pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda")
Herper::with_CondaEnv("eggnog",system("ete3 ncbiquery -v 0"),pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda")


#install_CondaTools("python=2.7","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)

#install_CondaTools("ete3_external_apps","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)

# install_CondaTools(c("eggnog-mapper","ete3","ete_toolchain","ete3_external_apps"),"eggnog",channels="etetoolkit", pathToMiniConda= "/rugpfs/fs0/brc/scratch/mpaul/herper_conda/")
# install_CondaTools("ete3","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)
# install_CondaTools("ete_toolchain","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)
#install_CondaTools("ete3_external_apps","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper/eggnog",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)

```



```{r}
read.delim("TC_ 20201201_Result/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227.emapper.annotations", skip = 3) %>% DT::datatable()

read.delim("TC_ 20201201_Result/eggnogAnno_WithPredictOrthp_CondaVersionPy2.emapper.predict_orthologs", skip = 3) %>% DT::datatable()
test <- read.delim("TC_ 20201201_Result/eggnogAnno_WithPredictOrthp_CondaVersionPy2.emapper.predict_orthologs")
nrow(test)

# seed_ortholog just shows first 4 columns of the annotations result
# read.delim("TC_ 20201201_Result/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227.emapper.seed_orthologs", skip = 5) %>% DT::datatable()


#We are missing the orhtolog file 

```

```{r, cache=T}
# comapre ant to bombyx mori 
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7091 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7091_FAisObiroi"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```


```{r, cache=T}
# comapre bombyx mori to ant
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/bombyx.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 2015173 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis2015173_FAisBmori"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```

```{r, cache=T}
# comapre bombyx mori to melanogaster
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/bombyx.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7227 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227_FAisBmori"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```

```{r, cache=T}

# comapre to ant to melanogaster
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7227 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227_FAisObiroi"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```


```{r, cache=T, eval =F}

Herper::with_CondaEnv("eggnog",system("python /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/eggnogConda/eggnog-mapper-refactor/emapper.py -i /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7227 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_refactor_Scopeis50557_Targetis7227"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```


```{R}
library(DBI)
library(dplyr)
library(dbplyr)
library(RSQLite)
library(tidyr)
library(stringr)

con <- DBI::dbConnect(RSQLite::SQLite(), dbname = "/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/eggnogConda/dataDirConda/eggnog.db")


#DBI::dbReadTable(con, 'eggnog') %>% filter(grep("28IP2@1",groups))
anno_list<-
  lapply(dir(".", pattern="FAisObiroi.emapper.annotations", full.names = T), read.delim, skip=3)
names(anno_list)<-c("7091","7227")
#my_anno <- anno_list[[1]]


#test %>% filter(grepl("42EW3@68295",groups))

#DBI::dbGetQuery(con, "select * from eggnog limit 10") %>% separate(name, c("Taxon", "Protein"), "[.]", remove = F) %>% filter()


## get db
mydb <- DBI::dbReadTable(con, "eggnog") %>% 
  separate(name, c("Taxon", "Protein"), "[.]", remove = F, extra = "merge") 

# mydb_7227 <- mydb %>% filter(Taxon=="7227")
# mydb_7091 <- mydb %>% filter(Taxon=="7091")


data_labels<-cbind("queryID"=c("2015173","2015173"), "queryname"=c("Obiroi","Obiroi"),"answerID"=c("7227","7091"),"answerName"=c("Dmelanogaster","Bmori")  )


my_homologs<- bplapply(1:length(anno_list), function(y, anno_list, data_labels, mydb){

my_current_db <-mydb %>% filter(Taxon==data_labels[y,3])
### do it for just one anno
my_anno <- anno_list[[y]]
## filter db down to just OGs of interest
test <- apply(my_anno,1,function(x){
  ogs<-str_split(x[19], ",", simplify = T)
  matches<-lapply(ogs, function(x){my_current_db %>% filter(grepl(x,groups)) %>% .[,1:3]})
  names(matches) <- ogs
  matches<-do.call( "rbind", matches)
  return(matches)
})
names(test) <- my_anno$X.query_name


## collapse down to equivalent names across ogs
OG_orthos<-lapply(1:length(test),function(x){
  my_name<-names(test)[x]
  #test[[x]]  %>% as_tibble(rownames="OG") %>% separate(OG,c("OG","n"),"[.]")
  
  if(length(test[[x]]$Protein)>0){
  orthos<-data.frame(my_name,unique(test[[x]]$Protein))
  }else{
  orthos<-data.frame(my_name,NA)  
  }
  
  colnames(orthos)<-c(paste(data_labels[y,1:2], collapse="_"),paste(data_labels[y,3:4], collapse="_"))
  
  return(orthos)
  })

OG_orthos<-do.call( "rbind", OG_orthos)

}, anno_list=anno_list, data_labels=data_labels, mydb=mydb)

comp_names<-apply(data_labels,1,function(x){paste(paste(x[1:2], collapse="_"),paste(x[3:4], collapse="_"),sep="vs")})

names(my_homologs)<-comp_names
rio::export(my_homologs, "obiroi_homolog_list.xlsx")

```



```{r}
require(rio)

#parse some of the datasets

#zhang tribo datsaset
zhang_pro<-import("published sets/1_zhang_JBC_2014/jbc.M114.599597-3_COMPARETOTRILBOIUM.xlsx")[-c(1,2),-11]
colnames(zhang_pro)<-c("Category", "Gene_Bm", "ID","LL","LP","PA","Gene_Tc", "ID", "RNAi_Pheno_Larva","RNAi_Pheno_Adult")

it<- zhang_pro %>% .$Category %>% is.na()
for (i in 1:length(it)){
if(it[i]==TRUE){
  it2<-max(which(!it)[which(!it)<i])
  zhang_pro$Category[i]<-zhang_pro$Category[it2]
}}
zhang_pro[3,c(2:4)]<-zhang_pro[2,c(2:4)]

zhang_pro[,4:6][is.na(zhang_pro[,4:6])]<-0
zhang_pro[,4:6][zhang_pro[,4:6]=="âˆš"]<-1

zhang_trib <- list()
zhang_trib$verified_RNAi<-rbind(zhang_pro[c(1:25,37:45),])
zhang_trib$unverified_RNAi<-zhang_pro[27:35,]
  
#zhang prot datsaset

zhang_pro<-import("published sets/1_zhang_JBC_2014/jbc.M114.599597-6_proteinlist.xls")
zhang_pro<-zhang_pro[-c(1:3),]
colnames(zhang_pro)<-c("LL","LP","PA","Gene_Bm","Accesion_ID", "Weight","pI","Tc", "Dm", "Ms", "Hs", "Notes")
zhang_pro<-zhang_pro[!is.na(zhang_pro$Accesion_ID),]

zhang_pro$Category <-c(rep("Immunity related protein",10), rep("Peptidase",9), rep("Oxido-dehydrogenase",19), rep("Hydrolase",13), rep("Esterase",5), rep("ATPase",3), rep("Serine protease",16), rep("Serpin",18), rep("Nutrition storage",12), rep("Binding protein",11), rep("Chitine related protein",8), rep("Cuticular protein",6), rep("Yellow protein",5), rep("Other protein",8), rep("Proteins with unknown function",29) )

zhang_pro[,1:3][is.na(zhang_pro[,1:3])]<-0
zhang_pro[,1:3][zhang_pro[,1:3]=="âˆš"]<-1

zhang_pro$Gene_Bm <- gsub("\n","",zhang_pro$Gene_Bm )
zhang_pro$Weight <- gsub("\n","",zhang_pro$Weight)
zhang_pro$Accesion_ID <- gsub("\n","",zhang_pro$Accesion_ID)
zhang_pro$pI <- gsub("\n","",zhang_pro$pI)

zhang_pro <- zhang_pro[,c(13,4,1,2,3,5:12)]


#liu prot datsaset
tmp<-import("published sets/2_Liu_JP_2018/1-s2.0-S1874391917304104-mmc1.xlsx", which="Table S2")

colnames(tmp)<-c(tmp[1,])
tmp<-tmp[-1,]
colnames(tmp)[1]<-"Category"
liu_allproteins<-tmp
  
tmp<-import("published sets/2_Liu_JP_2018/1-s2.0-S1874391917304104-mmc1.xlsx", which="Table S3")
my_names<-c(paste0("WMF_",tmp[2,1:6]), paste0("PMF_",tmp[2,1:6]))
tmp<-tmp[-(1:2),]
colnames(tmp)<-my_names
tmp1<-data.frame(tmp[,1:3],"LP"=1)
tmp2<-data.frame(tmp[,7:9],"PA"=1)

tmp3<-merge(liu_allproteins,tmp1[,c(1,4)], by.x="Protein ID", by.y="WMF_Protein.ID", all=T)
tmp3<-merge(tmp3,tmp2[,c(1,4)], by.x="Protein ID", by.y="PMF_Protein.ID", all=T)
tmp3[apply(tmp3[,c(23,24)], 1,function(x){all(is.na(x))}),c(23,24)]<-1
tmp3[,c(23,24)][is.na(tmp3[,c(23,24)])]<-0

liu_pro<-tmp3

liu_pro<-liu_pro[c(2,1,23,24,3:22)]

#Qu prot datsaset

# this dataset is a little weird. they did MS for both LP and PA stages, but they did not actually give info on either apart from a ratio. 
tmp<-import("published sets/3_Qu_JPR_2014/pr5000957_si_005 (1).xlsx", which="Table S2")

colnames(tmp)<- tmp[2,]
tmp<-tmp[-(1:2),]
qu_pro<-tmp[,c(2,3,1,4:16)]

```

Compare datasets
```{R}  
require(UpSetR)
require(viridis)
#start comparing datasets. integrate trib to start

#merge to a single ID. then merge them together. 


trib_in <- merge(data.frame("Gene_Bm"=unique(zhang_trib$verified_RNAi[,2]), "T_T"=1), zhang_pro, all=T)
#need some manula annotation as matches are not perfect due to human annotation
trib_in[c(15:17,105,64,116,129,163,176),2]<-1
trib_in<-trib_in[-c(14,23,63,88,98,162,175),]
trib_in$T_T[is.na(trib_in$T_T)]<-0

zhang_overlaps<-trib_in[,c(2,4,5,6)]
zhang_overlaps[]<-lapply(trib_in[,c(2,4,5,6)], function(x) as.numeric(as.character(x)))

#upset(zhang_overlaps)




bombyx <- UniProt.ws(7091)

NP<-trib_in %>% filter(str_detect(Accesion_ID, "^NP")) %>% pull(Accesion_ID) %>% AnnotationDbi::select(bombyx, keys=., keytype="REFSEQ_PROTEIN", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))

trib_in$Accesion_ID<- gsub("XP_004930585.1","BGIBMGA006497", trib_in$Accesion_ID)
trib_in$Accesion_ID<- gsub("-PA","",trib_in$Accesion_ID)

BG<-trib_in %>% filter(str_detect(Accesion_ID, "^BG")) %>% pull(Accesion_ID) %>% AnnotationDbi::select(bombyx, keys=., keytype="ENSEMBL_GENOMES", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))

EMBL<-trib_in %>% filter(!str_detect(Accesion_ID, "^BG")) %>% filter(!str_detect(Accesion_ID, "^NP")) %>%  pull(Accesion_ID)  %>% AnnotationDbi::select(bombyx, keys=., keytype="EMBL/GENBANK/DDBJ_CDS", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))
  
  
NP_merge<-trib_in %>% filter(str_detect(Accesion_ID, "^NP")) %>% merge(.,NP, by.x="Accesion_ID",  by.y= "REFSEQ_PROTEIN" ) #%>% filter(!duplicated(Accesion_ID))
colnames(NP_merge)[1]<-"REFSEQ_PROTEIN"

BG_merge<-trib_in %>% filter(str_detect(Accesion_ID, "^BG")) %>% merge(.,BG, by.x="Accesion_ID",  by.y= "ENSEMBL_GENOMES" )# %>% filter(!duplicated(Accesion_ID))
colnames(BG_merge)[1]<-"ENSEMBL_GENOMES"

EMBL_merge <-trib_in %>% filter(!str_detect(Accesion_ID, "^BG")) %>% filter(!str_detect(Accesion_ID, "^NP")) %>% merge(.,EMBL, by.x="Accesion_ID",  by.y= "EMBL/GENBANK/DDBJ_CDS" ) # %>% filter(!duplicated(Accesion_ID))
colnames(EMBL_merge)[1]<- "EMBL/GENBANK/DDBJ_CDS"

NP_merge <- NP_merge[,c(2,4:7,3,8:19,1)]
BG_merge <- BG_merge[,match(colnames(NP_merge), colnames(BG_merge))]
EMBL_merge <- EMBL_merge[,match(colnames(NP_merge), colnames(EMBL_merge))]


zhang_final <- rbind(NP_merge,BG_merge,EMBL_merge)[,c(1:6,14:19)]

colnames(zhang_final)[3:6] <-  paste0("Zha_",colnames(zhang_final)[3:6])
zhang_final <- data.frame(zhang_final, "Zha_any"=1)[,c(1:6,13,7:12)]
```


```{r}

liu_pro$`Protein ID` <- gsub("-PA","",liu_pro$`Protein ID`)
BG <- liu_pro  %>% filter(str_detect(`Protein ID`, "^BG"))
BG_merge <- BG %>%  pull(`Protein ID`)  %>% AnnotationDbi::select(bombyx, keys=.,  keytype="ENSEMBL_GENOMES", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))
BG_merge <- BG %>% merge(.,BG_merge, by.x= "Protein ID", by.y="ENSEMBL_GENOMES", all=T)
colnames(BG_merge)[1]<-"ENSEMBL_GENOMES"

liu_anno_nobg <- liu_pro  %>% filter(!str_detect(`Protein ID`, "^BG")) 
liu_anno_nobg[,"Protein ID"] <- liu_anno_nobg  %>% pull(`Protein ID`) %>% str_split(.,"[|]")  %>% sapply(., function(x){return(x[2])})
liu_anno <- liu_anno_nobg[,"Protein ID"] %>% AnnotationDbi::select(bombyx, keys=., keytype="GI_NUMBER*", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))
liu_anno_nobg <- liu_anno_nobg %>% merge(.,liu_anno, by.x= "Protein ID", by.y="GI_NUMBER*")
colnames(liu_anno_nobg)[1]<-"GI_NUMBER*"

liu_anno_nobg <- liu_anno_nobg[,match(colnames(BG_merge), colnames(liu_anno_nobg))]

liu_final <- rbind(liu_anno_nobg, BG_merge)[,c(7,2,3,4,26,27,1,28,29,25)]

colnames(liu_final)[c(1,2,5:10)]<-colnames(zhang_final)[c(1,2,8:13)]

colnames(liu_final)[3:4] <-  paste0("Liu_",colnames(liu_final)[3:4])
liu_final <- data.frame(liu_final, "Liu_any"=1)[,c(1:4,11,5:10)]

```


```{r}
# qu_pro$`Protein ID` <- gsub("gi|","",q_pro$`Protein ID`)
# qu_ids <- liu_pro %>% pull(`Protein ID`) %>% select(bombyx, keys=., keytype="GI_NUMBER*", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))


#  SILKDB ID
#  
#  ENSEMBL_GENOMES
# 
#  Accesion_ID (zhu) 
#  gi_number (qu)
 
```



```{R}

#start matches with DDBJ as fewest gaps
# apply(zhang_final,1, is.na) %>% rowSums()
# apply(liu_final,1, is.na) %>% rowSums()

liu_final[,6:11] <- apply(liu_final[,6:11], 1, str_trim) %>% t()
zhang_final[,8:13] <- apply(zhang_final[,8:13], 1, str_trim) %>% t()

ss<-unique(c(liu_final$EMBL.GENBANK.DDBJ, zhang_final$EMBL.GENBANK.DDBJ))[-1]
#456 and 168 unique ids

name_annos<-sapply(ss, function(x){
  
  ind_l<-which(liu_final$EMBL.GENBANK.DDBJ==x)
  ind_z<-which(zhang_final$EMBL.GENBANK.DDBJ==x)
  
  for (i in 1:2){
    ind<-get(c("ind_l","ind_z")[i])
    entry<-get(c("liu_final","zhang_final")[i])[ind,]
    grp<-c("Liu","Zhang")[i]
    
    if(length(ind)==0){
      my_out<- data.frame(
      NA,
      NA,
      t(as.data.frame(rep(0,5))),
      t(as.data.frame(rep(NA,7))))
      
    }else if(length(ind)==1){
    
    if(i==1){
      bin<-entry[3:5]
      my_names<-entry[6:11]
    }else if(i==2){
      bin<-entry[3:7]
      my_names<-entry[8:13]
    }
      
    my_out<-data.frame(
      entry[1],
      entry[2],
      bin,
      ind)
      
    }else{
      
      if(i==1){
      bin<-entry[1,3:5]
      my_names<-entry[,6:11]
        }else if(i==2){
      bin<-entry[1,3:7]
      my_names<-entry[,8:13]
        }
      
    my_out<-data.frame(
      entry[1,1],
      entry[1,2],
      bin,
      ind)
    }
    assign(paste0("my_out_",i), my_out)
    assign(paste0("my_names_",i), my_names)
  }

  my_names <-rbind(my_names_1, my_names_2) %>% apply(2,function(x){
    my_name<-unique(x)
    if(length(my_name)>1){
     my_name <- paste(my_name, collapse=";")
    }
    return(my_name)}
    )
  
  my_out<-data.frame(
      "Zhang_Name"=my_out_2[1,1],
      "Liu_Name"=my_out_1[1,1],
      "Zhang_Anno"=my_out_2[1,2],
      "Liu_Anno"=my_out_1[1,2],
      my_out_2[1,3:7],
      my_out_1[1,3:5],
      t(as.data.frame(my_names)),
      "l_count"=paste(my_out_1[,6], collapse=";"),
      "z_count"=paste(my_out_2[,8], collapse=";"))
  
  
  return(my_out)
  
})

name_annos_new <- as.data.frame(t(name_annos))

l_idx <- which(!is.na(name_annos_new$Liu_Name) & duplicated(name_annos_new$Liu_Name))
z_idx <- which(!is.na(name_annos_new$Zhang_Name) & duplicated(name_annos_new$Zhang_Name))

ph_fin<-rep(0,20)
idx_hit<-0

for (i in 1:length(l_idx)){
  print(i)
  idx_temp<-which(unlist(name_annos_new[,2])==unlist(name_annos_new[l_idx[i],2]))
  temp<-name_annos_new[idx_temp,]
  gi<-unique(temp$GI_NUMBER.) %>% unlist()
  
  temp2<-sapply(gi,function(y){
    gi_ind<-which(temp$GI_NUMBER. == y)
    temp_gi<-temp[gi_ind,]
    collapse_annos<-sapply(13:20, function(k){
    name1<-unique(temp_gi[!is.na(temp_gi[,k]),k])
    annos<-paste(name1[!(name1 == "NA")], collapse=";")
    return(annos)
      })
    collapse_annos<-sapply(collapse_annos, function(j){
    annos<- j %>% str_split(";", simplify=T) %>% unique() %>% paste(collapse=";")
    return(annos)
      })
    collapse_annos[collapse_annos==""] <- NA
    temp_out<-temp_gi[1,]
    temp_out[1,13:20]<-collapse_annos
    return(temp_out)
      })
  
   ph <- temp2 %>% t() %>% as.data.frame()  
   rownames(ph)<-NULL
   
   lapply(ph$GI_NUMBER., str_split, ";", simplify=T) -> test 
   test_flat<-test
   names(test_flat)<-letters[1:length(test_flat)]
   #test[[which(is.na(names(test)))]]<-NULL
   test_flat<-unlist(test_flat)
  
   my_overs<-sapply(1:length(test), function(b){
        
    my_overs<-test_flat[(test_flat %in% test[[b]]) & !is.na(names(test)[b])]
    my_overs<- names(my_overs) %>% gsub("[[:digit:]]", "", .)
    my_overs <- unique(my_overs[my_overs !=letters[b]])
    my_overs <- which(letters[1:26] %in% my_overs)
    my_out<-list(my_overs)
    names(my_out)<-b
    return(my_out)})
  
 my_overs[lengths(my_overs)==0 ]<-NA
  
 my_overs_am<-my_overs
 for(d in 1:length(my_overs)){
  z=1    
  if(any(!is.na(my_overs_am[[d]]))){
  while(z==1){

    for(k in 1:length(my_overs_am[[d]])){
      
      list_ind<-my_overs_am[[d]][k]
      partner<-my_overs_am[[list_ind]]
      if(any(!is.na(partner))){
        my_overs_am[[d]]<-c(my_overs_am[[d]],partner)
        my_overs_am[[list_ind]]<-NA
      }}
    my_overs_am[[d]] <- unique(my_overs_am[[d]])
    if( all(is.na(my_overs_am[(my_overs_am[[d]] %>% sort() %>% .[-1])]))){
      z<-0
    }
    }
    
  }}
  
  
my_overs_am<-my_overs_am[!is.na(my_overs_am)]
  
annot<-sapply(my_overs_am, function(r){
  annos<-sapply(13:20, function(k){
  annos<-ph[r,k] %>% unlist() %>% str_split(";", simplify=T) %>% unique() %>% sort() %>% paste(collapse=";")
  return(annos)
      }) %>% as.data.frame() %>% t()
  return(cbind(ph[r[1],1:12],annos))
  })  %>% t() %>% as.data.frame()

if(ncol(annot)==0){
ph_fin<-rbind(ph_fin,ph)  
  
}else{
colnames(annot)<-colnames(ph)
new_ph<-rbind(annot, ph[(-unlist(my_overs_am)),] )
ph_fin<-rbind(ph_fin,new_ph)
}


print(i)
}

ph_fin <- ph_fin[!is.na(unlist(ph_fin[,17])),]
ph_fin <- ph_fin[-1,]
colnames(ph_fin) <- colnames(name_annos_new)

library(magrittr)
ph_fin %<>% distinct() 

name_annos_liufixed<-rbind(name_annos_new[!name_annos_new$Liu_Name %in% unlist(ph_fin$Liu_Name),],ph_fin)

z_idx <- which(!is.na(name_annos_liufixed$Zhang_Name) & duplicated(name_annos_liufixed$Zhang_Name))

ph_fin<-rep(0,20)
idx_hit2<-0

for (i in 1:length(z_idx)){
  idx_temp<-which(unlist(name_annos_new[,2])==unlist(name_annos_new[z_idx[i],2]))
  temp<-name_annos_new[idx_temp,]
  gi<-unique(temp$GI_NUMBER.) %>% unlist()
  
  temp2<-sapply(gi,function(y){
    gi_ind<-which(temp$GI_NUMBER. == y)
    temp_gi<-temp[gi_ind,]
    collapse_annos<-sapply(13:20, function(k){
    name1<-unique(temp_gi[!is.na(temp_gi[,k]),k])
    annos<-paste(name1[!(name1 == "NA")], collapse=";")
    return(annos)
      })
    collapse_annos<-sapply(collapse_annos, function(j){
    annos<- j %>% str_split(";", simplify=T) %>% unique() %>% paste(collapse=";")
    return(annos)
      })
    collapse_annos[collapse_annos==""] <- NA
    temp_out<-temp_gi[1,]
    temp_out[1,13:20]<-collapse_annos
    return(temp_out)
      })
  
   ph <- temp2 %>% t() %>% as.data.frame()  
   rownames(ph)<-NULL
   
   lapply(ph$GI_NUMBER., str_split, ";", simplify=T) -> test 
   test_flat<-test
   names(test_flat)<-letters[1:length(test_flat)]
   #test[[which(is.na(names(test)))]]<-NULL
   test_flat<-unlist(test_flat)
  
   my_overs<-sapply(1:length(test), function(b){
        
    my_overs<-test_flat[(test_flat %in% test[[b]]) & !is.na(names(test)[b])]
    my_overs<- names(my_overs) %>% gsub("[[:digit:]]", "", .)
    my_overs <- unique(my_overs[my_overs !=letters[b]])
    my_overs <- which(letters[1:26] %in% my_overs)
    my_out<-list(my_overs)
    names(my_out)<-b
    return(my_out)})
  
 my_overs[lengths(my_overs)==0 ]<-NA
  
 my_overs_am<-my_overs
 for(d in 1:length(my_overs)){
  z=1    
  if(any(!is.na(my_overs_am[[d]]))){
  while(z==1){

    for(k in 1:length(my_overs_am[[d]])){
      
      list_ind<-my_overs_am[[d]][k]
      partner<-my_overs_am[[list_ind]]
      if(any(!is.na(partner))){
        my_overs_am[[d]]<-c(my_overs_am[[d]],partner)
        my_overs_am[[list_ind]]<-NA
      }}
    my_overs_am[[d]] <- unique(my_overs_am[[d]])
    if( all(is.na(my_overs_am[(my_overs_am[[d]] %>% sort() %>% .[-1])]))){
      z<-0
    }
    }
    
  }}
  
  
my_overs_am<-my_overs_am[!is.na(my_overs_am)]
  
annot<-sapply(my_overs_am, function(r){
  annos<-sapply(13:20, function(k){
  annos<-ph[r,k] %>% unlist() %>% str_split(";", simplify=T) %>% unique() %>% sort() %>% paste(collapse=";")
  return(annos)
      }) %>% as.data.frame() %>% t()
  return(cbind(ph[r[1],1:12],annos))
  })  %>% t() %>% as.data.frame()

if(ncol(annot)==0){
ph_fin<-rbind(ph_fin,ph)  
  
}else{
colnames(annot)<-colnames(ph)
new_ph<-rbind(annot, ph[(-unlist(my_overs_am)),] )
ph_fin<-rbind(ph_fin,new_ph)
}


print(i)
}








# for (i in 1:length(l_idx)){
#   idx_temp<-which(unlist(name_annos[,2])==unlist(name_annos[l_idx[i],2]))
#   temp<-name_annos[idx_temp,]
#   idx<-which(!is.na(temp[,1]) & !is.na(temp[,2]))
#   
#   idx_hit<-c(idx_hit,idx_temp)
#   
#   if(length(idx)>0){
#     ph<-temp[idx[1],]
#   } else {
#     ph<-temp[1,]
#   }
#   
#   collapse_annos<-sapply(13:20, function(j){
#     name1<-unique(temp[!is.na(temp[,j]),j])
#     annos<-paste(name1[!(name1 == "NA")], collapse=";")
#     
#     return(annos)
#       })
#   
#     collapse_annos<-sapply(collapse_annos, function(j){
#     annos<- j %>% str_split(";", simplify=T) %>% unique() %>% paste(collapse=";")
#     return(annos)
#       })
#   
#     ph[,13:20]<-collapse_annos
# 
#   ph_fin<-rbind(ph_fin,ph)
#   }



  #   
  # }else if(length(ind_l)==1 & length(ind_l)==1){
  #   z<-zhang_final[ind_z,]
  #   l<-liu_final[ind_l,]
  #   
  #   names_annos<-sapply(6:11, function(i){
  #     
  #     if(is.na(z[i+2])&is.na(l[i])){
  #       return(NA)
  #     }else if(is.na(z[i+2])){
  #       return(l[i])
  #     } else if(is.na(l[i])){
  #        return(z[i+2])
  #     } else if(z[i+2]==l[i]){
  #       return(z[i+2])
  #     }else{
  #       return(paste(z[i+2],l[i], sep=";"))
  #     }
  #     
  #     })
  # 
  #   names(names_annos)<-names(l)[6:11]
  #   
  # 
    
    
  #   
  #   }else{
  #     
  #   }
  # 
  # return(my_out)
  # })
  # 
  

```




````{r}
list_CondaPkgs("eggnog",pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

devtools::session_info()
```

