---
title: "20201201_OS_radierant_orthologs"
author: "Matt Paul"
date: "`r format(Sys.Date(), '%m/%d/%Y')`"
output:  
  html_document:
    code_folding: hide
    number_sections: TRUE
    theme: yeti
    highlight: tango
    toc: true
    toc_float: true
---

# Plan

Repeat comparison but include bombyx. 

compare and contrast the moori datasets. look for overlaps etc. 

Compare to d melanogaster.
then also comapre across in both directiosns.

Look for overlaps. 


retry refafctored in simplest way so i can post on github

```{r}
require(magrittr)
require(tidyverse)
require(ShortRead)
require(Herper)


```

```{r getSeqFromUniprot, eval =F}
require(UniProt.ws)
taxon <- availableUniprotSpecies(pattern="Bombyx")
bombyx <- UniProt.ws(7091)
allUniprotIDs <- keys(bombyx ,keytype = "UNIPROTKB")
allInfoColumns <- keytypes(bombyx )
grep("UNIPROTKB",
     allInfoColumns)

bgTable <- select(bombyx ,keys=allUniprotIDs,columns = allInfoColumns,keytype = "UNIPROTKB")
require(Biostrings)
seqToWrite <- AAStringSet(bgTable$SEQUENCE)
names(seqToWrite) <- bgTable$UNIPROTKB
writeXStringSet(seqToWrite,filepath = "/rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/bombyx.fa")

```


```{R, eval=F}

file.copy("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa", "obiroi.fa")
o_fa<-readFasta("obiroi.fa")

require(Herper)
export_CondaEnv("eggnog", yml_export= "eggnog_tom.yml", file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))
import_CondaEnv("eggnog_tom.yml", "eggnog", pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda")

#install_CondaTools("ete3_external_apps","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper/eggnog",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)
#external apps is not installed. this gives warning in build check. unclear why or if needed. 

Herper::with_CondaEnv("eggnog",system("ete3 build check"),pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda")
Herper::with_CondaEnv("eggnog",system("ete3 ncbiquery -v 0"),pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda")


#install_CondaTools("python=2.7","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)

#install_CondaTools("ete3_external_apps","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)

# install_CondaTools(c("eggnog-mapper","ete3","ete_toolchain","ete3_external_apps"),"eggnog",channels="etetoolkit", pathToMiniConda= "/rugpfs/fs0/brc/scratch/mpaul/herper_conda/")
# install_CondaTools("ete3","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)
# install_CondaTools("ete_toolchain","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper_conda",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)
#install_CondaTools("ete3_external_apps","eggnog",pathToMiniConda="/rugpfs/fs0/brc/scratch/mpaul/herper/eggnog",updateEnv=TRUE,channels="etetoolkit",verbose=TRUE)

```



```{r}
read.delim("TC_ 20201201_Result/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227.emapper.annotations", skip = 3) %>% DT::datatable()

read.delim("TC_ 20201201_Result/eggnogAnno_WithPredictOrthp_CondaVersionPy2.emapper.predict_orthologs", skip = 3) %>% DT::datatable()
test <- read.delim("TC_ 20201201_Result/eggnogAnno_WithPredictOrthp_CondaVersionPy2.emapper.predict_orthologs")
nrow(test)

# seed_ortholog just shows first 4 columns of the annotations result
# read.delim("TC_ 20201201_Result/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227.emapper.seed_orthologs", skip = 5) %>% DT::datatable()


#We are missing the orhtolog file 

```

```{r, cache=T}
# comapre to bombyx mori not melanogaster
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7091 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7091_FAisObiroi"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```


```{r, cache=T}
# comapre bombyx mori to melanogaster
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/bombyx.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 2015173 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis2015173_FAisBmori"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```

```{r, cache=T}
# comapre bombyx mori to melanogaster
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/bombyx.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7227 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227_FAisBmori"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```

```{r, cache=T}

# comapre to ant not melanogaster
Herper::with_CondaEnv("eggnog",system("emapper.py -i /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7227 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227_FAisObiroi"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```


```{r, cache=T, eval =F}

Herper::with_CondaEnv("eggnog",system("python /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/eggnogConda/eggnog-mapper-refactor/emapper.py -i /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/BRC/obiroi/obiroi.fa --cpu 30 -m diamond --tax_scope 50557 --target_taxa 7227 --resume --data_dir /rugpfs/fs0/brc/scratch/tcarroll/autoProcessing//eggnogConda/dataDirConda/ --output  /rugpfs/fs0/brc/scratch/mpaul/1_kronauer/orli/Kronauer_Orli/20201201_OS_eggnog/eggnogAnno_refactor_Scopeis50557_Targetis7227"),pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

```


```{R}
library(DBI)
library(dplyr)
library(dbplyr)
library(RSQLite)
library(tidyr)
library(stringr)

con <- DBI::dbConnect(RSQLite::SQLite(), dbname = "/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/eggnogConda/dataDirConda/eggnog.db")


#DBI::dbReadTable(con, 'eggnog') %>% filter(grep("28IP2@1",groups))

my_anno <- read.delim("TC_ 20201201_Result/eggnogAnno_CondaVersionPy2_Scopeis50557_Targetis7227.emapper.annotations", skip = 3)

#test %>% filter(grepl("42EW3@68295",groups))

#DBI::dbGetQuery(con, "select * from eggnog limit 10") %>% separate(name, c("Taxon", "Protein"), "[.]", remove = F) %>% filter()

mydb <- DBI::dbReadTable(con, "eggnog") %>% 
  separate(name, c("Taxon", "Protein"), "[.]", remove = F, extra = "merge") 

mydb_7227 <- mydb %>% filter(Taxon=="7227")
  
test <- apply(my_anno[1:211,],1,function(x){
  ogs<-str_split(x[19], ",", simplify = T)
  matches<-lapply(ogs, function(x){mydb_7227 %>% filter(grepl(x,groups)) %>% .[,1:3]})
  names(matches) <- ogs
  matches<-do.call( "rbind", matches)
  return(matches)
})

names(test) <- my_anno$X.query_name

OG_orthos<-lapply(1:length(test),function(x){
  my_name<-names(test)[x]
  #test[[x]]  %>% as_tibble(rownames="OG") %>% separate(OG,c("OG","n"),"[.]")
  orthos<-data.frame("query"=my_name,"result"=unique(test[[x]]$Protein))
  return(orthos)
  })

OG_orthos<-do.call( "rbind", OG_orthos)

save(OG_orthos,test,file = "ortho_tests.RData")

```



```{r}
require(rio)

#parse some of the datasets

#zhang tribo datsaset
zhang_pro<-import("published sets/1_zhang_JBC_2014/jbc.M114.599597-3_COMPARETOTRILBOIUM.xlsx")[-c(1,2),-11]
colnames(zhang_pro)<-c("Category", "Gene_Bm", "ID","LL","LP","PA","Gene_Tc", "ID", "RNAi_Pheno_Larva","RNAi_Pheno_Adult")

it<- zhang_pro %>% .$Category %>% is.na()
for (i in 1:length(it)){
if(it[i]==TRUE){
  it2<-max(which(!it)[which(!it)<i])
  zhang_pro$Category[i]<-zhang_pro$Category[it2]
}}
zhang_pro[3,c(2:4)]<-zhang_pro[2,c(2:4)]

zhang_pro[,4:6][is.na(zhang_pro[,4:6])]<-0
zhang_pro[,4:6][zhang_pro[,4:6]=="√"]<-1

zhang_trib <- list()
zhang_trib$verified_RNAi<-rbind(zhang_pro[c(1:25,37:45),])
zhang_trib$unverified_RNAi<-zhang_pro[27:35,]
  
#zhang prot datsaset

zhang_pro<-import("published sets/1_zhang_JBC_2014/jbc.M114.599597-6_proteinlist.xls")
zhang_pro<-zhang_pro[-c(1:3),]
colnames(zhang_pro)<-c("LL","LP","PA","Gene_Bm","Accesion_ID", "Weight","pI","Tc", "Dm", "Ms", "Hs", "Notes")
zhang_pro<-zhang_pro[!is.na(zhang_pro$Accesion_ID),]

zhang_pro$Category <-c(rep("Immunity related protein",10), rep("Peptidase",9), rep("Oxido-dehydrogenase",19), rep("Hydrolase",13), rep("Esterase",5), rep("ATPase",3), rep("Serine protease",16), rep("Serpin",18), rep("Nutrition storage",12), rep("Binding protein",11), rep("Chitine related protein",8), rep("Cuticular protein",6), rep("Yellow protein",5), rep("Other protein",8), rep("Proteins with unknown function",29) )

zhang_pro[,1:3][is.na(zhang_pro[,1:3])]<-0
zhang_pro[,1:3][zhang_pro[,1:3]=="√"]<-1

zhang_pro$Gene_Bm <- gsub("\n","",zhang_pro$Gene_Bm )
zhang_pro$Weight <- gsub("\n","",zhang_pro$Weight)
zhang_pro$Accesion_ID <- gsub("\n","",zhang_pro$Accesion_ID)
zhang_pro$pI <- gsub("\n","",zhang_pro$pI)

zhang_pro <- zhang_pro[,c(13,4,1,2,3,5:12)]


#liu prot datsaset
tmp<-import("published sets/2_Liu_JP_2018/1-s2.0-S1874391917304104-mmc1.xlsx", which="Table S2")

colnames(tmp)<-c(tmp[1,])
tmp<-tmp[-1,]
colnames(tmp)[1]<-"Category"
liu_allproteins<-tmp
  
tmp<-import("published sets/2_Liu_JP_2018/1-s2.0-S1874391917304104-mmc1.xlsx", which="Table S3")
my_names<-c(paste0("WMF_",tmp[2,1:6]), paste0("PMF_",tmp[2,1:6]))
tmp<-tmp[-(1:2),]
colnames(tmp)<-my_names
tmp1<-data.frame(tmp[,1:3],"LP"=1)
tmp2<-data.frame(tmp[,7:9],"PA"=1)

tmp3<-merge(liu_allproteins,tmp1[,c(1,4)], by.x="Protein ID", by.y="WMF_Protein.ID", all=T)
tmp3<-merge(tmp3,tmp2[,c(1,4)], by.x="Protein ID", by.y="PMF_Protein.ID", all=T)
tmp3[apply(tmp3[,c(23,24)], 1,function(x){all(is.na(x))}),c(23,24)]<-1
tmp3[,c(23,24)][is.na(tmp3[,c(23,24)])]<-0

liu_pro<-tmp3

liu_pro<-liu_pro[c(2,1,23,24,3:22)]

#Qu prot datsaset

# this dataset is a little weird. they did MS for both LP and PA stages, but they did not actually give info on either apart from a ratio. 
tmp<-import("published sets/3_Qu_JPR_2014/pr5000957_si_005 (1).xlsx", which="Table S2")

colnames(tmp)<- tmp[2,]
tmp<-tmp[-(1:2),]
qu_pro<-tmp[,c(2,3,1,4:16)]

```

comapre datasets
```{R}  
require(UpSetR)
require(viridis)
#start comparing datasets. integrate trib to start

#merge to a single ID. then merge them together. 


trib_in <- merge(data.frame("Gene_Bm"=unique(zhang_trib$verified_RNAi[,2]), "T_T"=1), zhang_pro, all=T)
#need some manula annotation as matches are not perfect due to human annotation
trib_in[c(15:17,105,64,116,129,163,176),2]<-1
trib_in<-trib_in[-c(14,23,63,88,98,162,175),]
trib_in$T_T[is.na(trib_in$T_T)]<-0

zhang_overlaps<-trib_in[,c(2,4,5,6)]
zhang_overlaps[]<-lapply(trib_in[,c(2,4,5,6)], function(x) as.numeric(as.character(x)))

upset(zhang_overlaps)




bombyx <- UniProt.ws(7091)

NP<-trib_in %>% filter(str_detect(Accesion_ID, "^NP")) %>% pull(Accesion_ID) %>% select(bombyx, keys=., keytype="REFSEQ_PROTEIN", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))

trib_in$Accesion_ID<- gsub("XP_004930585.1","BGIBMGA006497", trib_in$Accesion_ID)
trib_in$Accesion_ID<- gsub("-PA","",trib_in$Accesion_ID)

BG<-trib_in %>% filter(str_detect(Accesion_ID, "^BG")) %>% pull(Accesion_ID) %>% select(bombyx, keys=., keytype="ENSEMBL_GENOMES", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))

EMBL<-trib_in %>% filter(!str_detect(Accesion_ID, "^BG")) %>% filter(!str_detect(Accesion_ID, "^NP")) %>%  pull(Accesion_ID)  %>% select(bombyx, keys=., keytype="EMBL/GENBANK/DDBJ_CDS", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))
  
  
NP_merge<-trib_in %>% filter(str_detect(Accesion_ID, "^NP")) %>% merge(.,NP, by.x="Accesion_ID",  by.y= "REFSEQ_PROTEIN" ) %>% filter(!duplicated(Accesion_ID))
                                                              
BG_merge<-trib_in %>% filter(str_detect(Accesion_ID, "^BG")) %>% merge(.,BG, by.x="Accesion_ID",  by.y= "ENSEMBL_GENOMES PROTEIN" ) %>% filter(!duplicated(Accesion_ID))

EMBL_merge <-trib_in %>% filter(!str_detect(Accesion_ID, "^BG")) %>% filter(!str_detect(Accesion_ID, "^NP")) %>% merge(.,EMBL, by.x="Accesion_ID",  by.y= "EMBL/GENBANK/DDBJ_CDS" )  %>% filter(!duplicated(Accesion_ID))

liu_pro$`Protein ID` <- gsub("-PA",liu_pro$`Protein ID`)
liu_ids <- liu_pro %>% pull(`Protein ID`) %>% .[1:10] %>% select(bombyx, keys=., keytype="ENSEMBL_GENOMES", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))

qu_pro$`Protein ID` <- gsub("gi|","",q_pro$`Protein ID`)
qu_ids <- liu_pro %>% pull(`Protein ID`) %>% select(bombyx, keys=., keytype="GI_NUMBER*", columns=c("REFSEQ_PROTEIN","EMBL/GENBANK/DDBJ","EMBL/GENBANK/DDBJ_CDS","ENSEMBL_GENOMES","ENTREZ_GENE","GI_NUMBER*"))


SILKDB ID
 
 NSEMBL_GENOMES

 Accesion_ID (zhu) 
 gi_number (qu)
 
```



````{r}
list_CondaPkgs("eggnog",pathToMiniConda=file.path("/rugpfs/fs0/brc/scratch/tcarroll/autoProcessing/","eggnogConda"))

devtools::session_info()
```

